workflows/transport/pika_transport.py

```diff
@@ -504,14 +506,17 @@ def _broadcast(
                 routing_key="",
                 body=message,
                 properties=properties,
+                mandatory=True,
             )
-        except pika.exceptions.AMQPChannelError:
+        except pika.exceptions.AMQPChannelError as err:
             self._connected = False
-            raise
+            raise workflows.Disconnected(
+                "Caught a channel error: {}, stopping".format(err)
+            )
         except pika.exceptions.AMQPConnectionError:
             time.sleep(5)
+            self._connected = False
             if self.reconnect():
-                self._connected = False
                 self.reconnect()
```

Comment:
This looks like you are trying to connect twice as often as intended.
Should this `if` block just be the following?
```python
if not self.reconnect():
   raise workflows.Disconnected(
                    "Connection to RabbitMQ server was closed."
                )
```